File: WEEK 1/Coding Problems on InputOutput/solution.java
/*Here comes the big boy "JAVAAAA!!!!" */

/*Q1*/
import java.util.Scanner;

class UserInput {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter something: ");
        String userInput = scanner.nextLine();
        System.out.println("You entered: " + userInput);
        scanner.close();
    }
}

/*Q2*/

class Greeting{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your name: ");
        String name = scanner.nextLine();
        System.out.println("Hello, " + name + "!");
        scanner.close();
    }
}

/*Q3*/
class Address{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your city: ");
        String city = scanner.nextLine();
        System.out.print("Enter your state: ");
        String state = scanner.nextLine();
        System.out.println("Address: " + city + ", " + state);
        scanner.close();
    }
}

/*Q4*/
class AsteriskPattern{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the width: ");
        int width = scanner.nextInt();
        System.out.print("Enter the height: ");
        int height = scanner.nextInt();
        for (int i = 0; i < height; i++) {
            for (int j = 0; j < width; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
        scanner.close();
    }
}

File: WEEK 1/Coding Problems on InputOutput/solution.cpp
#include <iostream>
#include <string>
using namespace std;

//You can use cin also but using getline its better:))))

/*Q1*/
int main() {
    string user_input;
    cout << "Enter something: ";
    getline(cin, user_input);  
    cout << "You entered: " << user_input << endl;
    return 0;
}

/*Q2*/
int main() {
    string name;
    cout << "Enter your name: ";
    getline(cin, name);
    cout << "Hello, " << name << "!" << endl;
    return 0;
}

/*Q3*/
int main() {
    string city, state;
    cout << "Enter your city: ";
    getline(cin, city);
    cout << "Enter your state: ";
    getline(cin, state);
    cout << "Address: " << city << ", " << state << endl;
    return 0;
}

/*Q4*/
int main() {
    int width, height;
    cout << "Enter the width: ";
    cin >> width;
    cout << "Enter the height: ";
    cin >> height;
    for (int i = 0; i < height; i++) {
        for (int j = 0; j < width; j++) {
            cout << "*";
        }
        cout << endl;
    }
    return 0;
}